scripts/lib/CIME/XML/env_mach_specific.py

```diff
@@ -372,16 +372,17 @@ def get_mpirun(self, case, attribs, check_members=None, job="case.run"):
         the_match = best_match if best_match is not None else default_match
 
         # Now that we know the best match, compute the arguments
-        arg_node = self.get_optional_node("arguments", root=the_match)
-        if arg_node is not None:
-            arg_nodes = self.get_nodes("arg", root=arg_node)
-            for arg_node in arg_nodes:
-                arg_value = transform_vars(arg_node.text,
-                                           case=case,
-                                           subgroup=job,
-                                           check_members=check_members,
-                                           default=arg_node.get("default"))
-                args[arg_node.get("name")] = arg_value
+        if exe_only:
```

Comment:
This logic looks wrong - if exe_only is False then args is not assigned either?  I think you mean 
if not exe_only?